pkg/controller/scylladbmonitoring/sync_grafana.go

```diff
@@ -0,0 +1,502 @@
+			Required: requiredDeployment,
+			Control: resourceapply.ApplyControlFuncs[*appsv1.Deployment]{
+				GetCachedFunc: smc.deploymentLister.Deployments(sm.Namespace).Get,
+				CreateFunc:    smc.kubeClient.AppsV1().Deployments(sm.Namespace).Create,
+				UpdateFunc:    smc.kubeClient.AppsV1().Deployments(sm.Namespace).Update,
+				DeleteFunc:    smc.kubeClient.AppsV1().Deployments(sm.Namespace).Delete,
+			},
+		}.ToUntyped(),
+		resourceapply.ApplyConfig[*corev1.Service]{
+			Required: requiredService,
+			Control: resourceapply.ApplyControlFuncs[*corev1.Service]{
+				GetCachedFunc: smc.serviceLister.Services(sm.Namespace).Get,
+				CreateFunc:    smc.kubeClient.CoreV1().Services(sm.Namespace).Create,
+				UpdateFunc:    smc.kubeClient.CoreV1().Services(sm.Namespace).Update,
+				DeleteFunc:    smc.kubeClient.CoreV1().Services(sm.Namespace).Delete,
+			},
+		}.ToUntyped(),
+	}
+
+	if requiredIngress != nil {
+		applyConfigurations = append(applyConfigurations, resourceapply.ApplyConfig[*networkingv1.Ingress]{
+			Required: requiredIngress,
+			Control: resourceapply.ApplyControlFuncs[*networkingv1.Ingress]{
+				GetCachedFunc: smc.ingressLister.Ingresses(sm.Namespace).Get,
+				CreateFunc:    smc.kubeClient.NetworkingV1().Ingresses(sm.Namespace).Create,
+				UpdateFunc:    smc.kubeClient.NetworkingV1().Ingresses(sm.Namespace).Update,
+				DeleteFunc:    smc.kubeClient.NetworkingV1().Ingresses(sm.Namespace).Delete,
+			},
+		}.ToUntyped())
+	}
+
+	for _, cfg := range applyConfigurations {
+		// Enforce namespace.
+		cfg.Required.SetNamespace(sm.Namespace)
+
+		// Enforce labels for selection.
+		if cfg.Required.GetLabels() == nil {
+			cfg.Required.SetLabels(getGrafanaLabels(sm))
+		} else {
+			resourcemerge.MergeMapInPlaceWithoutRemovalKeys(cfg.Required.GetLabels(), getGrafanaLabels(sm))
+		}
+
+		// Set ControllerRef.
+		cfg.Required.SetOwnerReferences([]metav1.OwnerReference{
+			{
+				APIVersion:         scylladbMonitoringControllerGVK.GroupVersion().String(),
+				Kind:               scylladbMonitoringControllerGVK.Kind,
+				Name:               sm.Name,
+				UID:                sm.UID,
+				Controller:         pointer.Bool(true),
+				BlockOwnerDeletion: pointer.Bool(true),
+			},
+		})
+
+		// Apply required object.
+		_, changed, err := resourceapply.ApplyFromConfig(ctx, cfg, smc.eventRecorder)
+		if changed {
+			controllerhelpers.AddGenericProgressingStatusCondition(&progressingConditions, grafanaControllerProgressingCondition, cfg.Required, "apply", sm.Generation)
+		}
+		if err != nil {
+			gvk := resource.GetObjectGVKOrUnknown(cfg.Required)
+			applyErrors = append(applyErrors, fmt.Errorf("can't apply %s: %w", gvk, err))
+		}
+	}
+
+	cm := okubecrypto.NewCertificateManager(
+		smc.kubeClient.CoreV1(),
+		smc.secretLister,
+		smc.kubeClient.CoreV1(),
+		smc.configMapLister,
+		smc.eventRecorder,
+	)
+	for _, ccc := range certChainConfigs {
+		applyErrors = append(applyErrors, cm.ManageCertificateChain(
```

Comment:
make it consistnt with err collecting used everywhere else

Reply:
not sure I get the point, can you elaborate on what's wrong here?

Reply:
I meant the following:
```
err := cm.ManageCertificateChain(...)
if err != nil {
  applyErrors = append(applyErrors, fmt.Errorf("...", err))
}
```

Reply:
not applied

Reply:
done